src/server.rs

```diff
@@ -423,14 +423,14 @@ impl ServerState {
                         // Only send what a player has not yet seen
                         let raw_unsent_messages = match player.last_acked_msg_seq_num {
                             Some(last_acked_msg_seq_num) => {
-                                // Unwrap()'s okay because we know its non-empty
+                                // unwrap()'s okay because we know it's non-empty
                                 let mut message_iter = slot.messages.iter();
 
                                 let newest_msg = message_iter.clone().last().unwrap();
                                 // Player is caught up
                                 if last_acked_msg_seq_num == newest_msg.seq_num {
                                     continue;
-                                }
+                                } else if last_acked_msg_seq_num > newest_msg.seq_num { panic!("client has more messages than we sent!"); }
```

Comment:
This doesn't hold if we receive packets out of order, correct?

Reply:
@Manghi I think it would hold. This panics if `last_acked_msg_seq_num` is greater than `newest_msg.seq_num`. The former is the sequence number of the newest chat the client has received. The latter is the newest chat.

It's good that you drew attention to this, because I think it's a bug that we're panicking. This condition happens when the client is misbehaving, and that should be an error response to the client, not a server panic.

Reply:
I mixed a big indentation change with changing the panic to an error. Sorry! Please re-review @Manghi . Thanks :)

Reply:
Got it! Thanks for the clarity, didn't focus a lot in my haste.